### PR TITLE
lca: Adding oadpConent to ImageBasedUpgrade.

### PR DESCRIPTION
**Pull request by achuzhoy**
This aims at helping with bckup/restore automation.

### CODE REVIEW

**Comment by achuzhoy** [comment]
cc @trewest @cdvultur

File: pkg/lca/imagebasedupgrade.go [hunk 1]
```diff
@@ -223,6 +223,22 @@ func (builder *ImageBasedUpgradeBuilder) WithSeedImage(
...
+	oadpConentName string, oadpConentNamespace string) *ImageBasedUpgradeBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Appending oadpContent name %s in namespace %s to the imagebasedupgrade",
+		oadpConentName, oadpConentNamespace)
+
+	builder.Definition.Spec.OADPContent = append(builder.Definition.Spec.OADPContent,
+		lcav1alpha1.ConfigMapRef{Name: oadpConentName, Namespace: oadpConentNamespace})
```
Comment (trewest) [inline pkg/lca/imagebasedupgrade.go:237]:
```suggestion
	oadpContentName string, oadpContentNamespace string) *ImageBasedUpgradeBuilder {
	if valid, _ := builder.validate(); !valid {
		return builder
	}

	glog.V(100).Infof("Appending oadpContent name %s in namespace %s to the imagebasedupgrade",
		oadpContentName, oadpContentNamespace)

	builder.Definition.Spec.OADPContent = append(builder.Definition.Spec.OADPContent,
		lcav1alpha1.ConfigMapRef{Name: oadpContentName, Namespace: oadpContentNamespace})
```

File: pkg/lca/imagebasedupgrade.go [hunk 2]
```diff
@@ -223,6 +223,22 @@ func (builder *ImageBasedUpgradeBuilder) WithSeedImage(
 	return builder
 }
 
+// WithOadpContent adds oadpContent to be used by the imagebasedupgrade.
+func (builder *ImageBasedUpgradeBuilder) WithOadpContent(
+	oadpConentName string, oadpConentNamespace string) *ImageBasedUpgradeBuilder {
```
Comment (trewest) [inline pkg/lca/imagebasedupgrade.go:228]: WDYT about using parameter names like configMapName and configMapNamespace or some variation instead? It isn't clear that what you are passing in is a configmap reference based on the current argument names.

**Review by cdvultur — COMMENTED** [review]
Small comments and suggestions

File: pkg/lca/imagebasedupgrade.go (hunk 2 above)
Comment (achuzhoy) [reply pkg/lca/imagebasedupgrade.go]: Updated.

File: pkg/lca/imagebasedupgrade.go [hunk 3]
```diff
@@ -223,6 +223,22 @@ func (builder *ImageBasedUpgradeBuilder) WithSeedImage(
 	return builder
 }
 
+// WithOadpContent adds oadpContent to be used by the imagebasedupgrade.
+func (builder *ImageBasedUpgradeBuilder) WithOadpContent(
```
[the remark this answers is not in the document]
Comment (achuzhoy) [reply pkg/lca/imagebasedupgrade.go:228]: Updated.

File: pkg/lca/imagebasedupgrade.go (hunk 1 above)
Comment (achuzhoy) [reply pkg/lca/imagebasedupgrade.go]: Did you suggest something new? Seems the same.

File: pkg/lca/imagebasedupgrade.go [hunk 4]
```diff
@@ -223,6 +223,23 @@ func (builder *ImageBasedUpgradeBuilder) WithSeedImage(
 	return builder
 }
 
+// WithOadpContent adds oadpContent to be used by the imagebasedupgrade.
+// This is used for backup/restore during upgrade.
+func (builder *ImageBasedUpgradeBuilder) WithOadpContent(
+	oadpConentConfigMapName string, oadpConentConfigMapNamespace string) *ImageBasedUpgradeBuilder {
```
Comment (trewest) [inline pkg/lca/imagebasedupgrade.go:229]:
```suggestion
	oadpContentConfigMapName string, oadpContentConfigMapNamespace string) *ImageBasedUpgradeBuilder {
```

Comment (trewest) [reply pkg/lca/imagebasedupgrade.go:229]: @achuzhoy re: https://github.com/openshift-kni/eco-goinfra/pull/235#discussion_r1474633224, it's just a typo

Comment (trewest) [reply pkg/lca/imagebasedupgrade.go:229]: conent vs content

Comment (achuzhoy) [reply pkg/lca/imagebasedupgrade.go]: Updated. Thanks.

**Review by trewest — APPROVED** [review]
/lgtm

File: pkg/lca/imagebasedupgrade.go [hunk 5]
```diff
@@ -223,6 +223,23 @@ func (builder *ImageBasedUpgradeBuilder) WithSeedImage(
 	return builder
 }
 
+// WithOadpContent adds oadpContent to be used by the imagebasedupgrade.
+// This is used for backup/restore during upgrade.
+func (builder *ImageBasedUpgradeBuilder) WithOadpContent(
+	oadpContentConfigMapName string, oadpContentConfigMapNamespace string) *ImageBasedUpgradeBuilder {
```
Comment (kononovn) [inline pkg/lca/imagebasedupgrade.go:229]: No need to repeat argument type sting.
You can do just >
oadpContentConfigMapName, oadpContentConfigMapNamespace string

Comment (achuzhoy) [reply pkg/lca/imagebasedupgrade.go]: updated. thanks.

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**